packages/functions/index.test.js

```diff
@@ -0,0 +1,47 @@
+/* global globalThis */
+
+import { vi, expect, test } from 'vitest';
+
+import { waitUntil } from '.';
+
+test.each([
+  {},
+  () => {},
+  function () {},
+  NaN,
+  1,
+  false,
+  undefined,
+  null,
+  [],
+  '▲',
+])('waitUntil throws when called with a non-promise (%s)', input => {
+  expect(() => waitUntil(input)).toThrow(TypeError);
+  expect(() => waitUntil(input)).toThrow(
+    'waitUntil can only be called with a Promise'
```

Comment:
nit: Type errors should include the actual type we got to debug

```suggestion
    'waitUntil can only be called with a Promise, got ${typeof input}`
```